crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -1974,11 +1989,33 @@ fn conv_sort_path(
                 .sort_of_self_ty_alias(alias_to)?
                 .unwrap_or(rty::Sort::Err))
         }
-        fhir::SortRes::PrimSort(fhir::PrimSort::Set) => rty::SortCtor::Set,
-        fhir::SortRes::PrimSort(fhir::PrimSort::Map) => rty::SortCtor::Map,
+        fhir::SortRes::PrimSort(fhir::PrimSort::Set) => {
+            if path.args.len() != 1 {
+                // Has to have one argument
+                return Err(emit_prim_sort_generics_error(genv, path, "Set", 1))?;
+            }
+            rty::SortCtor::Set
+        }
+        fhir::SortRes::PrimSort(fhir::PrimSort::Map) => {
+            if path.args.len() != 2 {
+                // Has to have two arguments
+                return Err(emit_prim_sort_generics_error(genv, path, "Map", 2))?;
+            }
+            rty::SortCtor::Map
+        }
         fhir::SortRes::User { name } => rty::SortCtor::User { name },
```

Comment:
User defined opaque sort are a very obscure feature, but we should also give an error in a case like this:

```rust
extern crate flux_rs;

flux_rs::defs! {
    opaque sort MyOpaqueSort;
}

#[flux_rs::opaque]
#[flux_rs::refined_by(f: MyOpaqueSort<int>)]
struct S {}
```

User opaque sorts always have zero parameters

Reply:
@nilehmann, I think this would be caught by the type parameter expecting no generics?